### DISCUSSION TITLE
scaling issue

### DISCUSSION THREAD

**Discussion by cloudssky1997** [HaxeFlixel/flixel#2724] [Q&A]
Hi there. I was trying to perform simple UI animations in a FlxSpriteGroup that
consists of 2 buttons and a sprite using FlxTween. However I've  been facing an issue
when i want to scale the FlxSpriteGroup by a factor of 0.5 using FlxTween. It seems like
the FlxSpriteGroup does not update it's hitbox nor position. Please check the below images:

before scaling:
![Screenshot from 2023-02-01 21-55-06](https://user-images.githubusercontent.com/124206461/216194447-7095d10c-e6ff-4060-ae27-e1c1400ca4c5.png)

after scaling using FlxTween:
![Screenshot from 2023-02-01 21-53-48](https://user-images.githubusercontent.com/124206461/216194576-d8b4a3d4-3f6e-438a-965c-1bf9e35556e7.png)

How it's supposed to be:
![Screenshot from 2023-02-01 21-55-24](https://user-images.githubusercontent.com/124206461/216194635-3c56f10e-089e-4b71-a86e-d9bf700d1179.png)

Here is my code:
![Screenshot from 2023-02-02 03-51-36](https://user-images.githubusercontent.com/124206461/216194825-64d25cd2-bdb6-4b5e-a2cd-330a6cc78e99.png)

![Screenshot from 2023-02-02 03-51-55](https://user-images.githubusercontent.com/124206461/216194857-3080ce68-9dff-43c4-be10-ad84f87039c5.png)

Is there any way i could fix this? Thanks in advance.

**Comment by Geokureli** [comment]
Unfortunately FlxSpriteGroups don't work very well with scale. We're planning to over haul this class for flixel 6 because it's very hacky. You'll need to reposition the buttons manually after scaling

**Comment by mayo78** [comment]
for each member you can multiply their x and y by the scale and update their hitbox too
